azurerm/internal/services/frontdoor/frontdoor_custom_https_configuration_resource.go

```diff
@@ -0,0 +1,312 @@
+	}
+
+	return nil
+}
+
+func frontDoorCustomHttpsConfigurationReadParams(d *schema.ResourceData) (err error, resourceGroup string, frontDoorName string, frontendEndpointName string) {
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err, "", "", ""
+	}
+	resourceGroup = id.ResourceGroup
+	frontDoorName = id.Path["frontdoors"]
+	// Link to issue: https://github.com/Azure/azure-sdk-for-go/issues/6762
+	if frontDoorName == "" {
+		frontDoorName = id.Path["Frontdoors"]
+	}
+	frontendEndpointName = id.Path["frontendendpoints"]
+	if frontendEndpointName == "" {
+		frontDoorName = id.Path["FrontendEndpoints"]
+	}
+
+	return nil, resourceGroup, frontDoorName, frontendEndpointName
+}
+
+func resourceArmFrontDoorFrontendEndpointCustomHttpsConfigurationUpdate(ctx context.Context, frontendEndpointId string, customHttpsProvisioningEnabled bool, frontDoorName string, frontendEndpointName string, resourceGroup string, provisioningState frontdoor.CustomHTTPSProvisioningState, customHTTPSConfigurationCurrent *frontdoor.CustomHTTPSConfiguration, customHttpsConfigurationNew []interface{}, meta interface{}) error {
+	// Locking to prevent parallel changes causing issues
+	locks.ByID(frontendEndpointId)
+	defer locks.UnlockByID(frontendEndpointId)
+
+	if provisioningState != "" {
+		// Check to see if we are going to change the CustomHTTPSProvisioningState, if so check to
+		// see if its current state is configurable, if not return an error...
+		if customHttpsProvisioningEnabled != NormalizeCustomHTTPSProvisioningStateToBool(provisioningState) {
+			if err := IsFrontDoorFrontendEndpointConfigurable(provisioningState, customHttpsProvisioningEnabled, frontendEndpointName, resourceGroup); err != nil {
+				return err
+			}
+		}
+
+		if customHttpsProvisioningEnabled {
+			// Build a custom Https configuration based off the config file to send to the enable call
+			// NOTE: I do not need to check to see if this exists since I already do that in the validation code
+			customHTTPSConfiguration := customHttpsConfigurationNew[0].(map[string]interface{})
+			minTLSVersion := frontdoor.OneFullStopTwo // Default to TLS 1.2
+			if httpsConfig := customHTTPSConfigurationCurrent; httpsConfig != nil {
+				minTLSVersion = httpsConfig.MinimumTLSVersion
+			}
+			customHTTPSConfigurationUpdate := makeCustomHttpsConfiguration(customHTTPSConfiguration, minTLSVersion)
+			if provisioningState == frontdoor.CustomHTTPSProvisioningStateDisabled || customHTTPSConfigurationUpdate != *customHTTPSConfigurationCurrent {
+				// Enable Custom Domain HTTPS for the Frontend Endpoint
+				if err := resourceArmFrontDoorFrontendEndpointEnableHttpsProvisioning(ctx, true, frontDoorName, frontendEndpointName, resourceGroup, customHTTPSConfigurationUpdate, meta); err != nil {
+					return fmt.Errorf("Unable to enable/update Custom Domain HTTPS for Frontend Endpoint %q (Resource Group %q): %+v", frontendEndpointName, resourceGroup, err)
```

Comment:
Same here with casing...

```suggestion
					return fmt.Errorf("unable to enable/update Custom Domain HTTPS for Frontend Endpoint %q (Resource Group %q): %+v", frontendEndpointName, resourceGroup, err)
```